### PR TITLE
Added maven-shade-plugin to build uber jar with all dependencies

### LINKED ISSUE #10: Provide build-breaker as fat-jar
Because of https://github.com/oracle/graal/issues/3159 it is not possible for some customers to execute the build-breaker plugin, as they use Apline Linux 3.13+

Instead we should also provide this plugin as fat-jar with all dependencies, so that this can be executed in the pipeline

### PR DESCRIPTION
**Pull request by LumnitzF**
Fixes #10

### CODE REVIEW

**Comment by agoeb** [comment]
Great work! The jar builds, is usable on my machine, and works as expected. I just noticed a few things about the build itself that might need a little improvement:

- There are deprecation warnings about "actions/checkout" and "actions/upload-artifact". Please update the respective versions.
- The UPX action issues a deprecation warning about the "file" parameter and suggests using "files" instead. Could you maybe switch that as well?
- The "build-shaded" step produces an artifact called "target-shaded". To me, "shaded" seems like an implementation detail, and what I am looking for is a Jar. Could you rename the target to "target-jar" or "target-java" please?
- I'd like the Jar to be publicly downloadable, so it might be a good idea to include it into the "release" step. WDYT?

**Comment by LumnitzF** [comment]
Applied the changes you mentioned.
The final jar is also included in the release step, however I can't/don't know how to test it

**Review by agoeb — CHANGES_REQUESTED** [review]
Few further remarks, including an important one regarding job dependencies. 
Sorry if I come across a bit picky, but I suppose fixing stuff in the workflow is cheapest while you're at it anyways. :-)

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -40,7 +60,8 @@ jobs:
     name: Windows Build
     runs-on: windows-2019
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout git
+        uses: actions/checkout@v3
       - name: Set up GraalVM
         uses: ayltai/setup-graalvm@v1
```
[the remark this answers is not in the document]
Comment (LumnitzF) [reply .github/workflows/actions.yml:66]: Created #12

**Pull request merged**